Enforce frozen string literal true

gfyoung opened the pull request:
Bringing https://github.com/rubocop-hq/rubocop/pull/7435 over the finish line.

* Addressed merge conflicts
* Addressed error message comment from @bbatsov (https://github.com/rubocop-hq/rubocop/pull/7435#issuecomment-570307705)

cc @bbatsov @koic @pirj @parkerfinch

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

pirj commented:
What is the difference with #7435 in a nutshell?

https://github.com/rubocop-hq/rubocop/pull/7435#issuecomment-570307705 still stands, I believe this new implementation should replace the existing `always`, not to be added as another option.

Question is - what is more painful, to release an option that might be considered obsolete moving forward, or change the behaviour of existing option that would affect existing users with `frozen_string_literal: false`.

gfyoung commented:
@pirj : As described in the commit message, the difference between #7435 is that I fixed the merge conflicts and addressed @bbatsov comment about consistent error messages (https://github.com/rubocop-hq/rubocop/pull/7435#discussion_r362199898).

As for https://github.com/rubocop-hq/rubocop/pull/7435#issuecomment-570307705, that is a pretty substantive change you are proposing.  I don't necessarily disagree with the end goal, just not sure that I would want to do it so quickly.

Not sure how fast you guys move in terms of a breaking change like these, but I'll leave it to you maintainers to figure that out.

parkerfinch commented:
Thanks for pushing this forward @gfyoung!

parkerfinch commented on spec/rubocop/cop/style/frozen_string_literal_comment_spec.rb:
```diff
@@ -463,5 +463,715 @@
       RUBY
     end
+
+    context 'auto-correct' do
```
Since the original PR here was opened, @pirj [updated](https://github.com/rubocop-hq/rubocop/commit/6fd61ce075828ce7d8d82bb8f6e531e4289ebb89#diff-092a0140f3ad2219c13d0abdade6897b) this spec file to use the `expect_correction` helper. I think we should update the new specs to use that approach as well, which will obviate the `auto-correct` context. Does that make sense?

gfyoung replied:
Yep, that makes sense.  I'll refactor the tests to use that.

gfyoung commented:
@bbatsov @koic : If you can have a look.  I think all comments have been addressed.

bbatsov commented:
I think this will do for now. Thanks for bringing it over the finish line!
